### PR TITLE
Scottx611x/tool manager updates

### PR DESCRIPTION
**Pull request by scottx611x**
- Test refactoring
- Provide fields on `Tool` & `ToolDefinition` to dump json in rather than the `ast.literal_eval()` approach from before
- @mccalluc & @jkmarx: `tool_manager.tests.py`  has a lengthy diff, but most of it is just moving existing members into places that make more sense

### CODE REVIEW

File: refinery/tool_manager/utils.py [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 import os
 
+from bioblend.galaxy.client import ConnectionError
 from django.conf import settings
 from django.contrib import admin
 from django.db import transaction
-
-from bioblend.galaxy.client import ConnectionError
-from jsonschema import RefResolver, validate, ValidationError
+from jsonschema import RefResolver, ValidationError, validate
```
Comment (jkmarx) [inline refinery/tool_manager/utils.py]: is that from isort? caps before lowercase?

Comment (scottx611x) [inline refinery/tool_manager/utils.py]: Yes,  isort puts `class`es first in alph. order

File: refinery/tool_manager/tests.py [hunk 2]
```diff
@@ -1,64 +1,150 @@
...
+        )
+        self.tool_defs_view = ToolDefinitionsViewSet.as_view(
+            {
+                'get': 'list',
+                'post': 'create'
+            }
+        )
 
-        self.url_root = '/api/v2/tool_definitions/'
+        self.tools_url_root = '/api/v2/tools'
```
Comment (jkmarx) [inline refinery/tool_manager/tests.py]: trailing / ?

File: refinery/tool_manager/tests.py [hunk 3]
```diff
@@ -368,6 +444,54 @@ def test_list_list_pair_workflow_tool_def_generation(self):
...
+    def test_list_pair_list_workflow_tool_def_validation(self):
+        with open(
+                "{}/workflows/LIST:PAIR:LIST.json".format(TEST_DATA_PATH)
+        ) as f:
+            workflow_annotation = json.loads(f.read())
+            self.assertIsNone(
+                validate_tool_annotation(workflow_annotation)
+            )
+
+    def test_list_pair_list_workflow_tool_def_generation(self):
```
Comment (jkmarx) [inline refinery/tool_manager/tests.py]: a lot of assetEquals, perhaps break into assert the different types separately.

**Pull request merged**